src/jit-diff/jit-diff.cs

```diff
@@ -159,23 +224,340 @@ private void SetRID()
+                                if (!Directory.Exists(tryPlatformPath))
+                                {
+                                    continue;
+                                }
+                            }
+
+                            if (needCrossgen && (_diffRoot != null))
+                            {
+                                tryCrossgen = Path.Combine(_diffRoot, "bin", "Product", buildDirName, GetCrossgenExecutableName(_moniker));
+                                if (!File.Exists(tryCrossgen))
+                                {
+                                    continue;
+                                }
+                            }
+
+                            if (needTestTree && (_diffRoot != null))
+                            {
+                                tryTestPath = Path.Combine(_diffRoot, "bin", "tests", buildDirName);
+                                if (!Directory.Exists(tryTestPath))
+                                {
+                                    continue;
+                                }
+                            }
+
+                            // If we made it here, we've filled in all the defaults we needed to fill in.
+                            // Thus, we found an architecture/build combination that has an appropriate
+                            // Core_Root path (in the "diff" tree) and test_root.
+
+                            if (tryPlatformPath != null)
+                            {
+                                _platformPath = tryPlatformPath;
+                                needCoreRoot = false;
+                                Console.WriteLine("Using --core_root={0}", _platformPath);
+                            }
+                            if (tryCrossgen != null)
+                            {
+                                _crossgenExe = tryCrossgen;
+                                needCrossgen = false;
+                                Console.WriteLine("Using --crossgen={0}", _crossgenExe);
+                            }
+                            if (tryTestPath != null)
+                            {
+                                _testPath = tryTestPath;
+                                needTestTree = false;
+                                Console.WriteLine("Using --test_root={0}", _testPath);
+                            }
+
+                            if (build != "release")
+                            {
+                                Console.WriteLine();
+                                Console.WriteLine("Warning: it is best practice to use a release build for --core_root, --crossgen, and --test_root.");
+                                Console.WriteLine();
+                            }
+
+                            break;
+                        }
+
+                        if (!needCoreRoot && !needCrossgen && !needBasePath && !needDiffPath && !needTestTree)
```

Comment:
If you can find an appropriate basePath and diffPath on x64 but can't find one of coreRoot/crossgen/testTree on x64, or similarly if you can find coreRoot and crossgen and testTree on x64 but can't find one of basePath/diffPath, and you can find all 5 on x86, it looks like this logic would lock in the set it finds on x64 and then just find the others on x86 and then try to run using defaults with mismatched architectures.